In [254]:
import numpy as np
import pandas as pd

#raw data as a csv file
filepath = 'data/NBASTATS2023.csv' 

#display all rows/columns
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

#read csv into specific dataframe
df = pd.read_csv(filepath)

#shows how many players and fields we are working with

print(df.shape) 

#uncomment below to see full dataset
#print(df)

#check for missing values in our dataframe
#ensure dataset was formatted and parsed ok

missing_values = df.isnull().sum()

print(missing_values)

#retrieve values for a specific player
print("Bam Adebayo Stats \n")
df.loc[df['NAME'] == 'Bam Adebayo']




(609, 29)
RANK     609
NAME       0
TEAM       0
POS        0
AGE        0
GP         0
MPG        0
USG%       0
TO%        1
FTA        0
FT%        0
2PA        0
2P%        0
3PA        0
3P%        0
eFG%       3
TS%        3
PPG        0
RPG        0
APG        0
SPG        0
BPG        0
TPG        0
P+R        0
P+A        0
P+R+A      0
VI         0
ORtg      24
DRtg      23
dtype: int64
Bam Adebayo Stats 



RANK         NAME TEAM  POS   AGE  GP   MPG  USG%   TO%  FTA    FT%   2PA  \
2   NaN  Bam Adebayo  Mia  C-F  25.7  75  34.6  25.3  14.4  402  0.806  1102   

     2P%  3PA    3P%   eFG%    TS%   PPG  RPG  APG  SPG  BPG  TPG   P+R   P+A  \
2  0.545   12  0.083  0.541  0.592  20.4  9.2  3.2  1.2  0.8  2.5  29.6  23.6   

   P+R+A   VI   ORtg   DRtg  
2   32.8  9.8  116.0  105.6

CLEANING

We can see that rank shows as invalid(NaN) for all players(if we uncomment line 19) , so lets drop rank from our dataset

We also can do further cleaning of our dataset. In this case, players whos stats cannot be calculated for some fields(NaN) likely have played not enough games, or have not gotten >0 in a certain stat. We can trim our dataset by either filtering players out who have played very few minutes/games(option 1), or we can just remove any player that has a null value for any stat(option 2), or input filler numbers for our NaN values(option 3). If you dont know exactly how some of the advanced stats are calculated, option 3 is risky, so we will focus on the first two options

In [256]:
df = df.drop('RANK', axis=1)
#show new shape, should have 1 less column than previously
df.shape

(609, 28)

In [257]:


#Option 1
#removes players who ahve played less than 10 games and average less than 
#minutes per game
option_1_df = df.loc[(df['GP']>10) & (df['MPG']>5)]
print(option_1_df.shape)

missing_values = option_1_df.isnull().sum()

print(missing_values)



(503, 28)
NAME     0
TEAM     0
POS      0
AGE      0
GP       0
MPG      0
USG%     0
TO%      0
FTA      0
FT%      0
2PA      0
2P%      0
3PA      0
3P%      0
eFG%     0
TS%      0
PPG      0
RPG      0
APG      0
SPG      0
BPG      0
TPG      0
P+R      0
P+A      0
P+R+A    0
VI       0
ORtg     0
DRtg     0
dtype: int64


We can see that we have trimmed 106 players from our list using the minutes/games restriction, and have in fact gotten
rid of all the players with missing values. However, this is likely overkill as some players have played less than 10 games and less than 5 minutes per game while having analyzable values.


In [259]:
#Option 2
#remove only players who have an invalid value
o2_df= df.loc[(df['ORtg'].notnull()) & (df['eFG%'].notnull())]
print(o2_df.shape)
#we can add in all the other or clauses, but these 2 do the trick due to how these stats overlap

missing_values = o2_df.isnull().sum()

print(missing_values)

(585, 28)
NAME     0
TEAM     0
POS      0
AGE      0
GP       0
MPG      0
USG%     0
TO%      0
FTA      0
FT%      0
2PA      0
2P%      0
3PA      0
3P%      0
eFG%     0
TS%      0
PPG      0
RPG      0
APG      0
SPG      0
BPG      0
TPG      0
P+R      0
P+A      0
P+R+A    0
VI       0
ORtg     0
DRtg     0
dtype: int64


Now we can see that we have elimnated only 24 players and still have all analyzable values

In [261]:
#I would like to see players who score over 20 points per game
#lets store this as a new dataframe and we can do some transformations
pts_over_20 = o2_df.loc[o2_df['PPG'] >= 20]
print(pts_over_20.shape)

#From this we can see that there are 60 players in the 2022-23 NBA season who averaged over 20 points each game



(60, 28)


To do some more advanced filtering: 
I want to separate elite shooters(1) from strong defensive
players(2), from those who are neither(3), or both(4). We are going to create a new field to tag our players with, and set some conditions for each category. 

In order to qualify as an elite shooter:
- over 50 games played(large sample size,82 games is max)
- score over 20 points per game(ppg)
- High efficiency(Free throw% over 80%, Field goal percentage over 35%, True shooting % over 55%)

In order to qualify as a strong defensive player:
- over 50 games played
- 107 or lower defensive rating(league average is 110, lower is better)
- 1 block or steal per game




In [263]:
elite_shooters = pts_over_20.loc[(pts_over_20['GP']>50) & (pts_over_20['TS%']> .55) 
                                 & (pts_over_20['eFG%'] >.35) & (pts_over_20['FT%']>.80)]
print(elite_shooters.shape)
elite_shooters

(31, 28)


NAME TEAM  POS   AGE  GP   MPG  USG%   TO%  FTA  \
2                Bam Adebayo  Mia  C-F  25.7  75  34.6  25.3  14.4  402   
26              Desmond Bane  Mem    G  24.8  58  31.8  26.1  11.8  205   
53          Bojan Bogdanovic  Det    F  34.0  59  32.1  25.9  13.3  303   
57              Devin Booker  Pho    G  26.4  53  34.6  31.8  12.0  358   
77             Jalen Brunson  Nyk    G  26.6  68  35.0  27.2  10.6  392   
85              Jimmy Butler  Mia    F  33.6  64  33.4  25.6  10.2  555   
102          Jordan Clarkson  Uta    G  30.8  61  32.6  27.8  15.3  245   
116            Stephen Curry  Gol    G  35.1  56  34.7  31.0  13.6  281   
125            DeMar DeRozan  Chi  G-F  33.7  74  36.2  27.8  10.5  523   
154              Joel Embiid  Phi  C-F  29.1  66  34.6  37.0  14.5  771   
170           Darius Garland  Cle    G  23.2  69  35.5  26.9  15.0  322   
174              Paul George  Lac    F  32.9  56  34.6  29.5  14.9  294   
177  Shai Gilgeous-Alexander  Okc  G-F  24.7  68  35.5  32.8  12.2  739   
186             Jerami Grant  Por    F  29.1  63  35.6  22.8  11.3  342   
201        Tyrese Haliburton  Ind    G  23.1  56  33.6  23.9  14.4  203   
206             James Harden  Phi    G  33.6  58  36.8  25.0  18.8  361   
224              Tyler Herro  Mia    G  23.2  67  34.9  25.6  12.5  182   
270             Nikola Jokic  Den    C  28.1  69  33.7  27.2  19.5  415   
307              Zach LaVine  Chi  G-F  28.1  77  36.0  28.3  12.3  428   
314            Kawhi Leonard  Lac    F  31.8  52  33.6  27.0   9.2  279   
318           Damian Lillard  Por    G  32.7  58  36.3  33.8  13.7  558   
335          Lauri Markkanen  Uta  F-C  25.9  66  34.4  26.6  10.0  399   
345             Tyrese Maxey  Phi    G  22.4  60  33.6  24.1   8.1  213   
365         Donovan Mitchell  Cle    G  26.6  68  35.8  32.1  11.4  368   
379             Jamal Murray  Den    G  26.1  65  32.8  26.1  12.2  216   
422             Jordan Poole  Gol    G  23.8  82  30.0  29.2  16.5  415   
427       Kristaps Porzingis  Was  F-C  27.7  65  32.6  27.4  11.8  416   
483          Anfernee Simons  Por    G  23.8  62  35.0  25.1  10.9  179   
501             Jayson Tatum  Bos  F-G  25.1  74  36.9  32.7  12.0  622   
508            Klay Thompson  Gol    G  33.2  69  33.0  26.4   8.9  132   
574               Trae Young  Atl    G  24.6  73  34.8  32.6  17.7  639   

       FT%   2PA    2P%  3PA    3P%   eFG%    TS%   PPG   RPG   APG  SPG  BPG  \
2    0.806  1102  0.545   12  0.083  0.541  0.592  20.4   9.2   3.2  1.2  0.8   
26   0.883   532  0.534  407  0.408  0.568  0.606  21.5   5.0   4.4  1.0  0.4   
53   0.884   529  0.539  353  0.411  0.570  0.627  21.6   3.8   2.6  0.6  0.1   
57   0.855   751  0.554  316  0.351  0.546  0.601  27.8   4.5   5.5  1.0  0.3   
77   0.829   873  0.519  322  0.416  0.547  0.597  24.0   3.5   6.2  0.9  0.2   
85   0.850   785  0.564  103  0.350  0.560  0.647  22.9   5.8   5.3  1.8  0.3   
102  0.816   572  0.530  459  0.338  0.519  0.558  20.8   4.0   4.4  0.5  0.2   
116  0.915   494  0.579  639  0.427  0.614  0.656  29.4   6.1   6.3  0.9  0.4   
125  0.872  1161  0.526  142  0.324  0.522  0.592  24.5   4.6   5.1  1.1  0.5   
154  0.857  1128  0.587  200  0.330  0.573  0.655  33.1  10.2   4.2  1.0  1.7   
170  0.863   717  0.492  412  0.410  0.537  0.587  21.6   2.7   7.8  1.2  0.1   
174  0.871   578  0.521  426  0.371  0.536  0.588  23.8   6.1   5.1  1.5  0.4   
177  0.905  1213  0.533  168  0.345  0.531  0.626  31.4   4.8   5.5  1.6  1.0   
186  0.813   554  0.523  359  0.401  0.554  0.606  20.5   4.5   2.4  0.8  0.8   
201  0.867   439  0.572  402  0.400  0.586  0.624  20.7   3.7  10.4  1.6  0.4   
206  0.867   424  0.495  419  0.384  0.536  0.607  21.0   6.1  10.7  1.2  0.5   
224  0.934   572  0.497  537  0.378  0.531  0.566  20.1   5.4   4.2  0.8  0.2   
270  0.822   873  0.675  149  0.383  0.660  0.701  24.5  11.8   9.8  1.3  0.7   
307  0.848   844  0.556  544  0.375  0.558  0.607  24.8   4.5   4.2  0.9  0.2   
314

We can see we have a pool of 32 players who meet our criteria for elite shooters

In [265]:
strong_defense = o2_df.loc[o2_df['GP']>20]
strong_defense = strong_defense.loc[(strong_defense['DRtg']<107) 
                                    & ((strong_defense['SPG']>=1)|(strong_defense['BPG']>=1))]
print(strong_defense.shape)
strong_defense

(47, 28)


NAME TEAM  POS   AGE  GP   MPG  USG%   TO%  FTA    FT%  \
1           Steven Adams  Mem    C  29.7  42  27.0  14.6  23.1  129  0.364   
2            Bam Adebayo  Mia  C-F  25.7  75  34.6  25.3  14.4  402  0.806   
8          Jarrett Allen  Cle    C  25.0  68  32.6  16.5  12.9  221  0.733   
9          Jose Alvarado  Nor    G  25.0  61  21.5  19.8  14.2   80  0.813   
23              Mo Bamba  Orl    C  24.9  40  17.0  16.3   9.3   51  0.686   
45      Patrick Beverley  Chi    G  34.7  22  27.5  10.5  13.1   15  0.533   
51       Bismack Biyombo  Pho    C  30.6  61  14.3  14.0  19.8   70  0.357   
73          Jaylen Brown  Bos  G-F  26.5  67  35.9  31.5  12.5  344  0.765   
94           Alex Caruso  Chi    G  29.1  67  23.5  11.1  21.2   73  0.808   
103          Nic Claxton  Bro  F-C  24.0  76  29.9  15.5  14.1  246  0.541   
118        Anthony Davis  Lal  F-C  30.1  56  34.0  28.4  11.2  444  0.784   
154          Joel Embiid  Phi  C-F  29.1  66  34.6  37.0  14.5  771  0.857   
156       Bruno Fernando  Hou  F-C  24.7  31  11.7  15.1  17.7   44  0.682   
169       Daniel Gafford  Was  F-C  24.5  78  20.6  15.2  17.7  187  0.679   
179          Rudy Gobert  Min    C  30.8  70  30.7  16.3  18.3  340  0.644   
190       Draymond Green  Gol    F  33.1  73  31.5  13.2  30.1  108  0.713   
216            Josh Hart  Nyk    G  28.1  25  30.0  12.6  19.5   57  0.789   
252    Jaren Jackson Jr.  Mem  F-C  23.6  63  28.4  24.7  11.5  306  0.788   
254       Isaiah Jackson  Ind    F  21.2  63  16.5  17.5  14.0  146  0.651   
270         Nikola Jokic  Den    C  28.1  69  33.7  27.2  19.5  415  0.822   
275        Herbert Jones  Nor    F  24.5  66  29.5  14.5  14.9  161  0.764   
296     Christian Koloko  Tor    C  22.8  58  13.8  10.4  11.7   59  0.627   
297         John Konchar  Mem    G  27.0  72  20.7  10.4  10.4   36  0.778   
316         Caris LeVert  Cle    G  28.6  74  30.3  18.7  13.5  180  0.722   
323          Brook Lopez  Mil    C  35.0  78  30.4  19.3  10.6  190  0.784   
350      Jalen McDaniels  Cha  F-C  25.2  56  26.7  17.4  13.8  104  0.846   
357    De'Anthony Melton  Phi    G  24.9  77  27.9  16.5  13.5   87  0.793   
366          Evan Mobley  Cle    C  21.8  79  34.4  20.2  13.3  303  0.673   
402       Onyeka Okongwu  Atl  F-C  22.3  80  23.1  15.1  14.0  196  0.781   
405       Victor Oladipo  Mia    G  30.9  42  26.3  21.1  17.7   79  0.747   
420         Jakob Poeltl  Tor    C  27.5  26  27.2  17.5  11.2   72  0.569   
447        Nick Richards  Cha    C  25.4  65  18.7  15.7  15.8  183  0.749   
449      Josh Richardson  Nor    G  29.6  23  23.2  14.5  13.4   21  0.762   
453    Mitchell Robinson  Nyk  C-F  25.0  59  26.9  10.1  12.8  122  0.484   
481          Ben Simmons  Bro  G-F  26.7  42  26.3  14.3  29.2   57  0.439   
485         Marcus Smart  Bos    G  29.1  61  32.1  17.8  19.2  118  0.746   
497          Jalen Suggs  Orl    G  21.9  53  23.5  20.4  17.5  119  0.723   
501         Jayson Tatum  Bos  F-G  25.1  74  36.9  32.7  12.0  622  0.854   
512       Xavier Tillman  Mem    F  24.2  61  19.3  13.7  12.6   89  0.551   
523    Jarred Vanderbilt  Uta    F  24.0  52  24.1  14.1  17.9  102  0.657   
524    Jarred Vanderbilt  Lal    F  24.0  26  24.1  12.5  17.1   37  0.784   
548    Russell Westbrook  Lal    G  34.4  52  28.7  28.9  20.2  238  0.655   
556  Robert Williams III  Bos  C-F  25.5  35  23.5  11.8  16.6   41  0.610   
563        Mark Williams  Cha    C  21.3  43  19.3  17.0  13.7   97  0.691   
570       Christian Wood  Dal    F  27.5  67  25.9  26.4  13.6  281  0.772   
573       Thaddeus Young  Tor    F  34.8  54  14.7  13.5  17.5   26  0.692   
576          Ivica Zubac  Lac    C  26.1  76  28.5  14.8  18.5  238  0.697   

      2PA    2P%  3PA    3P%   eFG%    TS%   PPG   RPG  APG  SPG  BPG  TPG  \
1     262  0.599    1  0.000  0.597  0.564   8.6  11.5  2.3  0.9  1.1  1.9   
2    1102  0.545   12  0.083  0.541  0.592  20.4   9.2  3.2  1.2  0.8  2.5   
8     616  0.653   10  0.100  0.645  0.67

We have 47 players who meet our criteria as strong defenders

Next lets move our criteria to value for fantasy games. Fantasy basketball for the league I am in uses the following modifiers: 

Points Scored:1 Points

Total Rebounds:1.2 Points

Assists:1.5 Points

Steals:3 Points

Blocked Shots:3 Points

Turnovers:-1 Points

Missed shot = -.3 Points

(in order to accurately calculate efficiency based on points scored/missed shots, we will use a weighted formula based on accuracy of 2 pointers, 3 pointers, and freethrows)

In [307]:
#Lets create a fantasy points category and append it to our dataframe, maxing out at two decimal points

from math import trunc

#truncate our values so that its rounded to 2 decimals
def truncate(num):
    num = trunc(num*100)/100
    return num
fp = []

for index, row in o2_df.iterrows():
    fanpoints = truncate((1.2 * row['RPG']) + (1.5 * row['APG']) +
                         (3*(row['SPG'] +row['BPG'])) - row['TPG'] + 
                         #formula calculating points scored and missed shots
                         (2*row['2P%']) +(3*row['3P%']) -(.3*(1-(row['2P%'])+(1-row['3P%']))) +(row['FTA']*row['FT%']/row['GP']))
    
    fp.append(fanpoints)


  


In [328]:
o2_df['FANPOINTS'] = fp
o2_df.iloc[0]
#o2_df[['NAME','FANPOINTS']].sort_values(by='FANPOINTS', ascending = False)

/tmp/ipykernel_84/1056801953.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  o2_df['FANPOINTS'] = fp


NAME         Precious Achiuwa
TEAM                      Tor
POS                         F
AGE                      23.6
GP                         55
MPG                      20.7
USG%                     19.3
TO%                      12.7
FTA                       124
FT%                     0.702
2PA                       296
2P%                     0.564
3PA                       108
3P%                     0.269
eFG%                    0.521
TS%                     0.554
PPG                       9.2
RPG                       6.0
APG                       0.9
SPG                       0.6
BPG                       0.5
TPG                       1.1
P+R                      15.2
P+A                      10.1
P+R+A                    16.1
VI                        7.1
ORtg                    112.3
DRtg                    106.9
FANPOINTS               13.91
Name: 0, dtype: object

In [311]:
import pandas as pd

#raw data as a csv file
filepath = 'NBASTATS2023.csv' 

#display all rows/columns
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
rankings = pd.read_csv('data/FantasyPros_2023_Overall_NBA_ADP_Rankings.csv')

In [341]:
print(rankings.shape)
rankings.iloc[1]
rankings[['NAME','AVG']].sort_values(by='AVG', ascending = True)

(257, 8)


<bound method NDFrame.head of                          NAME    AVG
0                Nikola Jokic    1.0
1                 Luka Doncic    2.7
2       Giannis Antetokounmpo    3.3
3                 Joel Embiid    4.0
4                Jayson Tatum    4.7
5     Shai Gilgeous-Alexander    5.3
6               Stephen Curry    8.0
7           Tyrese Haliburton    8.3
8              Damian Lillard    9.3
9                Kevin Durant   10.7
10           Domantas Sabonis   12.0
11              Anthony Davis   14.0
12            Anthony Edwards   14.3
13                 Trae Young   14.3
14                LaMelo Ball   15.3
15               Devin Booker   15.7
16               LeBron James   16.3
17           Donovan Mitchell   19.0
18          Victor Wembanyama   20.0
19               Kyrie Irving   21.3
20              Mikal Bridges   21.7
21              Pascal Siakam   24.0
22               Jimmy Butler   25.0
23         Karl-Anthony Towns   26.7
24               De'Aaron Fox   29.0
25      

This is a table that contains the average pick each player is taken in fantasy drafts across different websites.

In fantasy basketball, higher average draft pick means that the player has higher value when you are picking(opponent may select). Lets combine our rankings and stats dataframes and create a new column for weighted fantasy value(Fantasy points per game /(1.002 ** AVG draft pick), because a earlier avg draft pick with same points is slightly higher priority.

We will do an inner join on the name of the player to combine our data from our two tables, and then add our final column, WFPG(Weighted Fan Points/Game)

In [338]:
weighted = o2_df.copy(deep = True)
print(weighted.shape)
weightedFP = []
rankings = rankings.rename(columns = {'Player':'NAME'})
weighted = weighted.drop_duplicates('NAME')
rankings = rankings.drop_duplicates('NAME')
#print(rankings.head)
result = rankings.merge(weighted, how = 'inner', on = ['NAME'])
print(result.shape)
#208 overlapping players
#print(result.iloc[0])


for index,row in result.iterrows():
    wfpg = ((row['GP']/82)*(row['FANPOINTS']))/(1.002**row['AVG'])
    weightedFP.append(wfpg)
    
result['WFPG'] = weightedFP
result.shape   
result[['NAME','WFPG','GP']].sort_values(by='WFPG', ascending = False)

(585, 29)
(217, 36)


NAME       WFPG  GP
0                Nikola Jokic  32.264495  69
10           Domantas Sabonis  30.964239  79
3                 Joel Embiid  28.018351  66
1                 Luka Doncic  27.298678  66
4                Jayson Tatum  25.899303  74
2       Giannis Antetokounmpo  25.332058  63
5     Shai Gilgeous-Alexander  25.108315  68
13                 Trae Young  24.008420  73
11              Anthony Davis  22.532529  56
29             Nikola Vucevic  22.473871  82
20              Pascal Siakam  22.027629  71
33              Julius Randle  21.863793  77
12            Anthony Edwards  21.525031  79
21               Jimmy Butler  21.405123  64
24                Bam Adebayo  21.203378  75
47                Evan Mobley  21.177122  79
36               James Harden  21.168984  58
69                Nic Claxton  20.405244  76
43              Fred VanVleet  20.248584  69
31              DeMar DeRozan  19.923781  74
37         Kristaps Porzingis  19.554060  65
56             Alperen Sengun  19.432454  75
48             Scottie Barnes  19.348791  77
7           Tyrese Haliburton  19.284362  56
8              Damian Lillard  19.259756  58
26            Dejounte Murray  19.086643  74
50                Josh Giddey  18.883335  76
23               De'Aaron Fox  18.818851  73
28          Jaren Jackson Jr.  18.474160  63
61                Rudy Gobert  18.446900  70
53                  Ja Morant  18.360212  61
16               LeBron James  18.055352  55
95             Draymond Green  17.655011  73
45             Paolo Banchero  17.632574  72
57             Walker Kessler  17.470542  74
80                Brook Lopez  17.270382  78
17           Donovan Mitchell  17.181086  68
49               Jrue Holiday  17.169422  67
34             Darius Garland  17.065565  69
25            Lauri Markkanen  16.797643  66
27              Jalen Brunson  16.645047  68
76          Jonas Valanciunas  16.477461  79
46                Zach LaVine  16.410329  77
59              Jarrett Allen  16.367933  68
6               Stephen Curry  16.264764  56
30               Jaylen Brown  15.949579  67
81                 Chris Paul  15.913381  59
54               Myles Turner  15.891195  62
62                CJ McCollum  15.855227  75
51              Deandre Ayton  15.366962  67
121              Kevon Looney  15.293081  82
38               Jordan Poole  15.087135  82
44                Paul George  15.045848  56
42               Jamal Murray  14.678263  65
66             Jalen Williams  14.581452  75
109            Onyeka Okongwu  14.537792  80
60               Franz Wagner  14.450844  80
32              Kawhi Leonard  14.375693  52
119               Ivica Zubac  14.367971  76
100             Derrick White  14.317346  82
82               Clint Capela  14.297521  65
15               Devin Booker  14.062260  53
99               Aaron Gordon  13.814128  68
84           Jabari Smith Jr.  13.730663  79
102                 Tre Jones  13.674633  68
63                 Tyus Jones  13.573647  80
70                Jalen Green  13.345217  76
9                Kevin Durant  13.175059  39
88                Buddy Hield  13.161232  80
79              Tobias Harris  13.047249  74
141             Kyle Anderson  13.035350  69
58                 Kyle Kuzma  12.979655  64
19              Mikal Bridges  12.935133  56
40               Desmond Bane  12.831673  58
74               Terry Rozier  12.650583  63
64                Tyler Herro  12.574586  67
124         Immanuel Quickley  12.522899  81
107                Jaden Ivey  12.464931  74
111         Mitchell Robinson  12.388142  59
115            Christian Wood  12.340625  67
89               John Collins  12.229245  71
156          Kevin Porter Jr.  12.149707  59
87               Marcus Smart  12.134709  61
72               Jerami Grant  12.043329  63
96             Markelle Fultz  12.013456  60
188             Killian Hayes  11.997716  76
105               Jalen Duren  11.727102  67
152            T.J. McConnell  11.661997  75
125         De'Antho

Our rankings data contains 258 players, and not all of them match the data in our previous dataframe so in this case where our data is mismatched size, we will compute weighted fan point values only for those players that are contained in both datasets(217).  

The index for each player shows their original placement based on the rankings data(0 based indexing), we can see how rankings have shifted around based on our league's point calculations

We have now merged our dataframes, and created an effective ranking system for my fantasy draft!
